### PR TITLE
Add varchar to real coercion for hive tables

### PR DESCRIPTION
**Pull request by Praveen2112**
## Description

Add varchar to float coercion for hive tables.

Enable the Hive users to retrieve the information from their tables after performing queries which change varchar/ string columns to float.

Hive sample DDL query:

ALTER TABLE mytable CHANGE COLUMN mycolumn mycolumn float;

## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
(x) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
# Hive
* Add varchar to float coercion for hive tables. ({issue}`issuenumber`)
```

### CODE REVIEW

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/coercions/TestVarcharToFloatCoercer.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        assertVarcharToFloatCoercion("123456789.12345678", Float.parseFloat("123456789.12345678"));
+        assertVarcharToFloatCoercion("123", Float.parseFloat("123"));
+        assertVarcharToFloatCoercion("1.17549435E-38f", MIN_NORMAL); // Value where Float.intBitsToFloat(0x00800000)
+        assertVarcharToFloatCoercion("3.4028235e+38f", MAX_VALUE); // largest positive value
+        assertVarcharToFloatCoercion("-3.4028235e+38f", -3.4028235E38f); // largest negative value
+        assertVarcharToFloatCoercion("4.4028235e+39f", POSITIVE_INFINITY); // Value above max positive value
+        assertVarcharToFloatCoercion("-3.4028235e+39f", NEGATIVE_INFINITY); // Value below max negative value
+        assertVarcharToFloatCoercion("1.4e-45f", MIN_VALUE); // smallest positive nonzero value
+        assertVarcharToFloatCoercion("1.3e-46f", 0f); // Value smaller than positive nonzero value
+        assertVarcharToFloatCoercion("-1.3e-46f", -0f); // Value greater than negative nonzero value
```
Comment (krvikash) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/coercions/TestVarcharToFloatCoercer.java:51]: can we add a test for something like `12.3d` and with invalid `12.3e`?

**Review by krvikash — APPROVED** [review]

**Comment by Praveen2112** [comment]
@krvikash Thanks for review. Have added additional test cases

**Review by vlad-lyutenko — APPROVED** [review]

**Review by krvikash — APPROVED** [review]

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/coercions/CoercionUtils.java [hunk 2]
```diff
@@ -110,6 +110,9 @@ public static Type createTypeFromCoercer(TypeManager typeManager, HiveType fromH
         if (fromType instanceof VarcharType varcharType && toHiveType.equals(HIVE_DOUBLE)) {
             return Optional.of(new VarcharToDoubleCoercer(varcharType, coercionContext.treatNaNAsNull()));
         }
+        if (fromType instanceof VarcharType varcharType && toHiveType.equals(HIVE_FLOAT)) {
```
Comment (findinpath) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/coercions/CoercionUtils.java:113]: potential follow-up: There's a lot of `if` statements which deal with

```
fromType instanceof VarcharType varcharType
```

Maybe group them inside a big if would make the code easier the read. This is just an opinion, it doesn't need necessarily to be implemented.

**Review by findinpath — APPROVED** [review]

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/coercions/CoercionUtils.java (hunk 2 above)
Comment (Praveen2112) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/coercions/CoercionUtils.java:113]: I'm planning to perform a minor clean-up post the coercions

**Comment by Praveen2112** [comment]
@findinpath Thanks a lot for the review. Added changes in the documentation as well.

File: docs/src/main/sphinx/connector/hive.md [hunk 3]
```diff
@@ -642,7 +642,7 @@ type conversions.
 * - `BOOLEAN`
   - `VARCHAR`
 * - `VARCHAR`
-  - `TINYINT`, `SMALLINT`, `INTEGER`, `BIGINT`, `DOUBLE`, `TIMESTAMP`, `DATE`, as well as
+  - `TINYINT`, `SMALLINT`, `INTEGER`, `BIGINT`, `REAL`, `DOUBLE`, `TIMESTAMP`, `DATE`, as well as
```
Comment (findinpath) [inline docs/src/main/sphinx/connector/hive.md:645]: :+1:

**Pull request merged**